engine/src/main/java/org/pentaho/di/trans/steps/rest/Rest.java

```diff
@@ -0,0 +1,572 @@
+      }
+
+      // add status to output
+      if ( !Utils.isEmpty( data.resultCodeFieldName ) ) {
+        newRow = RowDataUtil.addValueData( newRow, returnFieldsOffset, new Long( status ) );
+        returnFieldsOffset++;
+      }
+
+      // add response time to output
+      if ( !Utils.isEmpty( data.resultResponseFieldName ) ) {
+        newRow = RowDataUtil.addValueData( newRow, returnFieldsOffset, new Long( responseTime ) );
+        returnFieldsOffset++;
+      }
+      // add response header to output
+      if ( !Utils.isEmpty( data.resultHeaderFieldName ) ) {
+        newRow = RowDataUtil.addValueData( newRow, returnFieldsOffset, headerString );
+      }
+    } catch ( Exception e ) {
+      throw new KettleException( BaseMessages.getString( PKG, "Rest.Error.CanNotReadURL", data.realUrl ), e );
+    } finally {
+      if ( webResource != null ) {
+        webResource = null;
+      }
+      if ( client != null ) {
+        client.destroy();
+      }
+    }
+    return newRow;
+  }
+
+  private void setConfig() throws KettleException {
+    if ( data.config == null ) {
+      // Use ApacheHttpClient for supporting proxy authentication.
+      data.config = new DefaultApacheHttpClient4Config();
+      if ( !Utils.isEmpty( data.realProxyHost ) ) {
+        // PROXY CONFIGURATION
+        data.config.getProperties().put( ApacheHttpClient4Config.PROPERTY_PROXY_URI, "http://" + data.realProxyHost + ":" + data.realProxyPort );
+        if ( !Utils.isEmpty( data.realHttpLogin ) && !Utils.isEmpty( data.realHttpPassword ) ) {
+          AuthScope authScope = new AuthScope( data.realProxyHost, data.realProxyPort );
+          UsernamePasswordCredentials credentials = new UsernamePasswordCredentials( data.realHttpLogin, data.realHttpPassword );
+          CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+          credentialsProvider.setCredentials( authScope, credentials );
+          data.config.getProperties().put( ApacheHttpClient4Config.PROPERTY_CREDENTIALS_PROVIDER, credentialsProvider );
+        }
+      } else {
+        if ( !Utils.isEmpty( data.realHttpLogin ) ) {
+          // Basic authentication
+          data.basicAuthentication = new HTTPBasicAuthFilter( data.realHttpLogin, data.realHttpPassword );
+        }
+      }
+      if ( meta.isPreemptive() ) {
+        data.config.getProperties().put( ApacheHttpClient4Config.PROPERTY_PREEMPTIVE_BASIC_AUTHENTICATION, true );
+      }
+      // SSL TRUST STORE CONFIGURATION
+      if ( !Utils.isEmpty( data.trustStoreFile ) ) {
+        try ( FileInputStream trustFileStream = new FileInputStream( data.trustStoreFile ) ) {
+          KeyStore trustStore = KeyStore.getInstance( "JKS" );
+          trustStore.load( trustFileStream, data.trustStorePassword.toCharArray() );
+          TrustManagerFactory tmf = TrustManagerFactory.getInstance( "SunX509" );
+          tmf.init( trustStore );
+
+          SSLContext ctx = SSLContext.getInstance( "SSL" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

An old version of SSL/TLS is enabled. We recommend that you enable TLS version 1.2, 1.3, or later. [Learn more](https://cheatsheetseries.owasp.org/cheatsheets/TLS_Cipher_String_Cheat_Sheet.html)